docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc

```diff
@@ -2,18 +2,23 @@
 
 When you build your pipeline, you might want to add tasks that require **secrets** in order to access external resources.
 
-NOTE: One such task is the link:https://github.com/konflux-ci/build-definitions/tree/main/task/sast-snyk-check[sast-snyk-check] task that uses the third-party service link:https://snyk.io/[snyk] to perform static application security testing (SAST) as a part of the default {ProductName} pipeline. Use this procedure to upload your snyk.io token. Name the secret `sast_snyk_task` so that the snyk task in the {ProductName} pipeline will recognize it and use it.
+.Secrets fall into three catagories of when they can be added:
+
+. Before a component is added. If a secret is needed to access the source control platform (like xref:creating-secrets.adoc#creating-source-control-secrets[GitLab]), then the secret must be created before the component is created.
+. Before a build succeeds. Some artifact build tasks need specific secrets to be able to pull all of the content to include in the final artifact. These secrets like for xref:creating-secrets.adoc#creating-registry-pull-secrets[container registries] can be added after the component is created but must be provided before a successful build can occur.
+. After a component has been onboarded. These secrets are often used in tasks. The tasks included in the Konflux pipelines will not fail if a secret is not created properly. Instead, the task will just not run the code like with xref:creating-secrets.adoc#creating-task-input-secrets[snyk].
```

Comment:
About xrefs.
Paths need to be given from the _pages_ family directory, e.g.
xref:how-tos/configuring/creating-secrets.adoc#creating-source-control-secrets[GitLab]

OR
Here we can also use in-page xrefs because we refer to the titles on the same page. This format should work (no spaces around a comma):

<<creating-source-control-secrets,GitLab>>
<<creating-registry-pull-secrets,container registries>>
<<creating-task-input-secrets,snyk>>

If Antora doesn't process them correctly for some reason, we can add ID's anywhere on the page like [#any-id-name] and then refer to it via <<any-id-name,text to show>>.